### PR TITLE
feat(fetch): support baseURL, add tests for invalid args

### PR DESCRIPTION
**Pull request by yury-s**
#5999

### CODE REVIEW

**Review by dgozman — APPROVED** [review]

File: tests/browsercontext-fetch.spec.ts [hunk 1]
```diff
@@ -349,3 +349,44 @@ it('should propagate extra http headers with redirects', async ({context, server
   expect(req3.headers['my-secret']).toBe('Value');
 });
 
+it('should throw on ivalid header value', async ({context, server}) => {
```
Comment (mxschmitt) [inline tests/browsercontext-fetch.spec.ts:352]:
```suggestion
it('should throw on invalid header value', async ({context, server}) => {
```

File: src/server/fetch.ts [hunk 2]
```diff
@@ -47,8 +47,7 @@ export async function playwrightFetch(context: BrowserContext, params: types.Fet
       agent = new HttpsProxyAgent(proxyOpts);
     }
 
-    // TODO(https://github.com/microsoft/playwright/issues/8381): set user agent
-    const fetchResponse = await sendRequest(context, new URL(params.url), {
+    const fetchResponse = await sendRequest(context, new URL(params.url, context._options.baseURL), {
```
Comment (mxschmitt) [inline src/server/fetch.ts:50]: Let's use this function maybe? https://github.com/microsoft/playwright/blob/master/src/utils/utils.ts#L343

**Review by mxschmitt — APPROVED** [review]

File: src/server/fetch.ts (hunk 2 above)
Comment (yury-s) [reply src/server/fetch.ts:50]: It returns a string and swallows error, we need neither of those here.

File: tests/browsercontext-fetch.spec.ts (hunk 1 above)
Comment (yury-s) [reply tests/browsercontext-fetch.spec.ts:352]: fixed

**Pull request merged**